tests/test_optimization.py

```diff
@@ -116,3 +116,241 @@ def test_tantrix_area(self):
+        self.barqcurve.initialize_parameters(
+            fit_target_curve=reference_curve,
+            fit_n_samples=100
+        )
+        
+        # Check that fitting produces reasonable results
+        fitted_points = self.barqcurve.get_bezier_control_points()
+        
+        # The fitted curve should be similar in overall characteristics
+        # (exact match not expected due to different n_free_points and optimization)
+        self.assertEqual(fitted_points.shape[1], 3, "Control points should be 3D")
+        self.assertGreater(fitted_points.shape[0], 8, "Should have sufficient control points")
+        
+        # Verify curve endpoints are reasonable (should start/end near origin for closed curve)
+        self.assertLess(jnp.linalg.norm(fitted_points[0]), 1e-6, "Curve should start at origin")
+        self.assertLess(jnp.linalg.norm(fitted_points[-1]), 1e-6, "Curve should end at origin")
+    def test_point_array_fitting_convergence(self):
+        """
+        Test fitting to discrete points and verify convergence behavior.
+        """
+        # Define target points
+        t_vals = jnp.linspace(0, 1, 10)
+        target_points = jnp.array([
+            [jnp.sin(2 * jnp.pi * t), jnp.cos(2 * jnp.pi * t), 0.1 * t] 
+            for t in t_vals
+        ])
+        
+        # Modified fitting function that tracks losses
+        def tracking_fit_func(target_curve=None, target_points=None, n_samples=100):
+            # Call original function WITH loss tracking enabled
+            result = self.barqcurve._fit_curve_to_control_points(
+                target_curve=target_curve, 
+                target_points=target_points, 
+                n_samples=n_samples,
+                track_losses=True  # ← Enable tracking
+            )
+            
+            if isinstance(result, tuple):
+                free_points, losses = result
+                return free_points, losses
+            return result, []
+        
+        # Initialize with point fitting and get loss history
+        fitted_points, loss_history = tracking_fit_func(target_points=target_points)
+        
+        # Verify basic properties
+        self.assertEqual(fitted_points.shape, (8, 3))
+        self.assertFalse(jnp.allclose(fitted_points, 0.0))
+        
+        # ← NUEVO: Verify convergence behavior
+        if len(loss_history) > 1:
+            # Check that loss generally decreases (allowing some fluctuation)
+            final_loss = loss_history[-1]
+            initial_loss = loss_history[0]
+            self.assertLess(final_loss, initial_loss * 1.5, 
+                        "Loss should decrease or at least not increase significantly")
+            
+            # Check that optimization is actually changing the loss
+            self.assertGreater(len(set(loss_history)), 1, 
+                            "Loss should change during optimization")
+
+    def test_error_handling_and_parameter_validation(self):
```

Comment:
Similarly, a lot of these tests are handled by other existing tests. You only need to keep Test 3 and Test 4 here, since those check the updated function signatures.